handlers/TileMapHandler.js

```diff
@@ -244,6 +246,44 @@ class TileMapHandler {
         }
     }
 
+    resetPlayerStartPoint() {
+        const levelObjects = this.levelObjects;
+        const startFlagsInLevel = levelObjects.filter(levelObject => levelObject.type === ObjectTypes.START_FLAG);
+        const doorsInLevel = levelObjects.filter(levelObject => levelObject.type === ObjectTypes.DOOR);
+        const defaultStartFlag = startFlagsInLevel.find(startFlag => startFlag.extraAttributes?.defaultStartFlag);
+        const fallbackStartFlag = startFlagsInLevel[startFlagsInLevel.length - 1];
+        let destinationData = PlayMode.customExit; // NOTE: customExit might be null/undefined
+        let destinationObject = null;
+
+        if (destinationData?.flagIndex) {
+            if (destinationData.type === ObjectTypes.DOOR) {
+                destinationObject = doorsInLevel.find((door) => {
+                    return door.extraAttributes?.flagIndex === PlayMode.customExit.flagIndex;
+                });
+            } else if (!destinationData.type || destinationData.type === ObjectTypes.START_FLAG) {
+                destinationObject = startFlagsInLevel.find((startFlag) => {
+                    return startFlag.extraAttributes?.flagIndex === PlayMode.customExit.flagIndex
+                });
+            }
```

Comment:
here you could have a longer if condition, and if flags and doors are in 1 array, you could iterate over this array and find flagIndex